File: patterns/c-patterns/pattern128.c
// Pattern 128: ...

//  ** **
// *  *  *
// *     *
//  *   *
//   * *
//    *

#include <stdio.h>

int main()
{
    int row, col;
    for (row = 0; row < 6; row++)
    {
        for (col = 0; col < 7; col++)
        {
            if ((row == 0 && col % 3 != 0) || (row == 1 && col % 3 == 0) || (row - col == 2) || (row + col == 8))
            {
                printf("*");
            }
            else
            {
                printf(" ");
            }
        }
        printf("\n");
    }
    return 0;
}

File: patterns/javascript-patterns/pattern128.js
// Pattern 128: ...

//  ** **
// *  *  *
// *     *
//  *   *
//   * *
//    *

for (let row = 0; row < 6; row++) {
    let rowOutput = '';
    for (let col = 0; col < 7; col++) {
        if ((row === 0 && col % 3 !== 0) || (row === 1 && col % 3 === 0) || (row - col === 2) || (row + col === 8)) {
            rowOutput += '*';
        } else {
            rowOutput += ' ';
        }
    }
    console.log(rowOutput);
}
